example/road-traffic/road-traffic-generators/src/main/java/uk/gov/gchq/gaffer/traffic/generator/RoadTrafficElementGenerator.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.gchq.gaffer.traffic.generator;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLogPlus;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.time.DateUtils;
+
+import uk.gov.gchq.gaffer.commonutil.CollectionUtil;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.data.generator.OneToManyElementGenerator;
+import uk.gov.gchq.gaffer.types.FreqMap;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public abstract class RoadTrafficElementGenerator<T> implements OneToManyElementGenerator<T> {
+
+    protected List<Entity> createCardinalities(final List<Edge> edges) {
+        final List<Entity> cardinalities = new ArrayList<>(edges.size() * 2);
+
+        for (final Edge edge : edges) {
+            cardinalities.add(createCardinality(edge.getSource(), edge.getDestination(), edge));
+            cardinalities.add(createCardinality(edge.getDestination(), edge.getSource(), edge));
+        }
+
+        return cardinalities;
+    }
+
+    protected Entity createCardinality(final Object source,
+                                     final Object destination,
+                                     final Edge edge) {
+        final HyperLogLogPlus hllp = new HyperLogLogPlus(5, 5);
+        hllp.offer(destination);
+
+        return new Entity.Builder()
+                .vertex(source)
+                .group("Cardinality")
+                .property("edgeGroup", CollectionUtil.treeSet(edge.getGroup()))
+                .property("hllp", hllp)
+                .property("count", 1L)
+                .build();
+    }
+
+    protected Date getDate(final String dCountString, final String hour) {
+        Date dCount = null;
+        try {
+            dCount = new SimpleDateFormat("dd/MM/yyyy HH:mm").parse(dCountString);
+        } catch (final ParseException e) {
+            // incorrect date format
```

Comment:
If we're seeing incorrect date formats, is it not worth logging this/reporting it to the user?  Otherwise it's going to look like data is loading, where actually it's being dropped because of the timestamp

Reply:
Agreed that this is a potential issue - I'll add an exception here if it was not possible to parse one of the date strings.